### PR TITLE
Transactions and blocks states

### PR DESCRIPTION
**Pull request by humb1t**
Signed-off-by: Nikita Puzankov <humb1t@yandex.ru>

### CODE REVIEW

File: iroha/src/sumeragi.rs [hunk 1]
```diff
@@ -10,17 +17,17 @@ use std::{
...
 }
 
 #[derive(Io, Decode, Encode, Debug, Clone)]
 pub enum Message {
     /// Is sent by leader to all validating peers, when a new block is created.
-    Created(Block),
+    Created(SignedBlock),
     /// Is sent by validating peers to proxy tail and observing peers when they have signed this block.
-    Signed(Block),
+    Signed(SignedBlock),
```
Comment (eadventurous) [inline iroha/src/sumeragi.rs:28]: The naming seems to be a bit controversial here. The block is in the process of being signed - it is not signed by all members yet.

File: iroha/src/sumeragi.rs [hunk 2]
```diff
@@ -48,9 +55,10 @@ pub struct Sumeragi {
     sorted_peers: Vec<PeerId>,
     max_faults: usize,
     peer_id: PeerId,
-    /// Block in discussion this round
-    pending_block: Option<Block>,
+    /// PendingBlock in discussion this round
```
Comment (eadventurous) [inline iroha/src/sumeragi.rs:58]: Shouldn't it be `SignedBlock in discussion ...` here?

File: iroha/src/sumeragi.rs [hunk 3]
```diff
@@ -68,45 +83,74 @@ impl Sumeragi {
...
+                    .into_iter()
+                    .map(|tx| tx.sign(Vec::new()))
+                    .filter_map(Result::ok)
+                    .collect(),
+            )
+            .build()
+            .sign(&self.public_key, &self.private_key)?;
+            let minimum_quorum_of_peers = 2;
+            if self.sorted_peers.len() < minimum_quorum_of_peers {
+                Ok(Some(block))
```
Comment (eadventurous) [inline iroha/src/sumeragi.rs:108]: Maybe it should also `kura.store` the block inside of the sumeragi. This behavior is a bit misleading: if there is only one peer it returns the block and stores it in `lib.rs`, if more peers it stores peer in `sumeragi.rs` .

File: iroha/src/sumeragi.rs (hunk 1 above)
Comment (humb1t) [reply iroha/src/sumeragi.rs:28]: It means it's signed at least once - when all signs are in place, should we name it "Voted" or smthg?

Comment (eadventurous) [reply iroha/src/sumeragi.rs:28]: Ok, I think we can leave it as it.

File: iroha/src/sumeragi.rs (hunk 2 above)
Comment (humb1t) [reply iroha/src/sumeragi.rs:58]: https://soramitsu.atlassian.net/browse/RH2-86

File: iroha/src/sumeragi.rs (hunk 3 above)
Comment (humb1t) [reply iroha/src/sumeragi.rs:108]: https://soramitsu.atlassian.net/browse/RH2-87

**Pull request merged**